apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1817,42 +1842,48 @@ close_solo_tx(Account, Nonce, #data{ on_chain_id = ChanId
                                      , nonce      => Nonce }),
     Ok.
 
-tx_defaults(channel_create_tx = Tx, #{acct := Initiator} = Opts,
-            #data{opts = DOpts}) ->
-    Nonce = try maps:get(nonce, Opts, maps:get(nonce, DOpts))
-            catch
-                error:_ ->
-                    {ok, N} =
-                        aec_next_nonce:pick_for_account(Initiator),
-                    N
-            end,
-    #{ fee   => default_fee(Tx)
-     , nonce => Nonce };
-tx_defaults(channel_deposit_tx = Tx, Opts, D) ->
-    Acct = my_account(D),
-    ChanId = cur_channel_id(D),
-    Opts1 = maps:merge(
-              tx_defaults(channel_create_tx, Opts#{acct => Acct}, D),
-              #{ fee        => default_fee(Tx)
-               , channel_id => ChanId }),
-    TTL = maps:get(ttl, Opts1, 0),
-    Opts1#{ttl => adjust_ttl(TTL)};
-tx_defaults(channel_withdraw_tx = Tx, Opts, D) ->
-    maps:merge(
-      tx_defaults(channel_deposit_tx, Opts, D),
-      #{ fee => default_fee(Tx) });  % same as deposit defaults
-tx_defaults(channel_slash_tx, _Opts, _D) ->
+tx_defaults(Type, Opts, #data{ on_chain_id = ChanId } = D) ->
+    Default = tx_defaults_(Type, Opts, D),
+    Default#{ channel_id => ChanId
+            , fee => default_fee(Type)
+            , nonce => default_nonce(Opts, D) }.
+
+tx_defaults_(channel_create_tx, _Opts, _D) ->
     #{};
-tx_defaults(channel_close_solo_tx = Tx, _Opts, _D) ->
-    #{ fee => default_fee(Tx) }.
+tx_defaults_(channel_deposit_tx = Tx, Opts, D) ->
+    Opts1 = tx_defaults_(channel_create_tx, Opts, D),
+    default_ttl(Tx, Opts1, D);
+tx_defaults_(channel_withdraw_tx, Opts, D) ->
+    tx_defaults_(channel_deposit_tx, Opts, D); % same as deposit defaults
+tx_defaults_(channel_slash_tx = Tx, Opts, D) ->
+    default_ttl(Tx, Opts, D);
+tx_defaults_(channel_close_solo_tx = Tx, Opts, D) ->
+    default_ttl(Tx, Opts, D);
+tx_defaults_(channel_close_mutual_tx, _Opts, _D) ->
+    #{}.
+
+default_nonce(Opts, #data{opts = DOpts}) ->
+    try maps:get(nonce, Opts, maps:get(nonce, DOpts))
+    catch
+        error:_ ->
+            Account = maps:get(acct, Opts),
+            ok(aec_next_nonce:pick_for_account(Account))
+    end.
 
+%% Note that the default fee will be used as a base for adjustment, once
+%% we have an actual transaction record (required by aetx:min_fee/2).
+%% The default should err on the side of being too low.
+-ifdef(TEST).
```

Comment:
I see removal of this tracked as https://www.pivotaltracker.com/story/show/164103197
Ok.